includes/class-wc-countries.php

```diff
@@ -656,12 +662,12 @@ public function get_default_address_fields() {
 				'priority'     => 50,
 			),
 			'address_2'  => array(
-				'placeholder'  => esc_attr__( 'Apartment, suite, unit etc. (optional)', 'woocommerce' ),
+				/* translators: use to display optional text in address2 field. */
+				'placeholder'  => sprintf( esc_attr__( 'Apartment, suite, unit etc. %s', 'woocommerce' ), esc_html( $address_2_palceholder ) ),
```

Comment:
Move this string into your conditional above.

Reply:
Done.